app/questionnaire/questionnaire_store_updater.py

```diff
@@ -160,6 +149,13 @@ def remove_list_item_data(self, list_name: str, list_item_id: str) -> None:
 
         self._progress_store.remove_progress_for_list_item_id(list_item_id=list_item_id)
 
+    def remove_list_data(self, list_name: str) -> None:
+        """Delete entire list and remove any associated answers"""
+        self._answer_store.remove_all_answers_for_list_item_ids(
+            *self._list_store[list_name].items
+        )
+        self._list_store.delete_list(list_name)
+
     def capture_dependencies_for_list_change(self, list_name: str) -> None:
```

Comment:
In this method, there is a scenario in which a KeyError is raised.

```
section_ids.update(
            self._schema.list_collector_section_ids_by_list_name[list_name]
        )
```

This needs to be updated as this would key error if the schema had no list collectors.

Can you ensure all methods called from `set_supplementary_data` are graceful.

Reply:
We can sort this in a new PR given it is existing, as this is need before Roofing Tiles schema can be launched.